### DISCUSSION TITLE
ERCOT Public API Client - 60-Day COP All Updates

### DISCUSSION THREAD

**Discussion by GlidePathAPI** [ercot/api-specs#40] [Q&A] [answered]
Hello,

I'm attempting to access data from the following API endpoint: https://api.ercot.com/api/public-reports/np3-991-ex/60_cop_all_updates. However, I am not receiving any data when I try to retrieve information for January 1st, 2023.

Steps to Reproduce
Send a GET request to the endpoint: https://api.ercot.com/api/public-reports/np3-991-ex/60_cop_all_updates.
Attempt to retrieve data specifically for January 1st, 2023.

Expected Behavior
The API should return data for the specified date range, including January 1st, 2023.

Actual Behavior
No data is returned when attempting to retrieve data for January 1st, 2023.

Additional Information
Is there a limit to the date range for this API?
Are there any known issues or restrictions related to accessing data for January 1st, 2023?

**Comment by cimview** [comment]
The earliest I see in 60-day COP in the API is 10/12/2023.  I am seeing a similar thing in the API for "LMPs by Settlement Points, ResourceNodes, and Trading Hubs" report - I see data going back to 12/31/2023 but nothing before that.  I don't think all of the data is available in the API.

If you can consume zip files, you might have more success with the Data Access Portal.   It looks like they have 60-day COP going back to February 2019.  https://data.ercot.com/data-product-archive/NP3-991-EX

**Reply by GlidePathAPI** [reply]
What method did you use to find the furthest date available? 

By consume zip files, do you mean a manual download process or some other means?

**Reply by cimview** [reply, marked as answer]
I just did a quick manual binary search using the API, plus a few dummy checks for random days before and after the cutoff to make sure that it really is a cutoff. 

Yes: The Data Access Portal above is designed for manual download of historical zip files you would have gotten if you were doing a daily data scrape.  Not sure what your use case is, though.

**Reply by GlidePathAPI** [reply]
Understood. Thank you for the help. 

Compiling the data for resource analysis. Manual would take a long time and was hoping it could be accomplished in a more efficient manner.

**Comment by mackermann-ercot** [comment]
Hello GlidePathAPI, 

Thanks for joining the GitHub discussions!

Our primary objective for the Public API project is to provide historical archives for our public EMIL products that are no longer available through our main ercot.com site. The file archive provides EMIL products for the last seven years (and in some cases much longer), however we only provide API access to the data within those products from mid December of 2023 year (during the Beta release).

However, If there is enough interest to provide additional historical data for selected reports, we may consider loading data prior to the beta release.

Regards,
mackermann, ERCOT

**Comment by GlidePathAPI** [comment]
Mackermann -

Thank you for the email.

I’m new to data analysis and learning API, web scraping, etc. so I apologize if the questions do not provide enough information or are incorrectly formed. Could you point me in the direction of a report that would allow me to see resource status and submitted high sustained limits through January of 2021?


Sean Ashman
Asset Management
C: 512.627.8966
Teams: 630.884.9154
***@***.******@***.***>
800 Town and Country Blvd, Suite 500
Houston, TX 77024

[A picture containing text, tableware, dishware  Description automatically generated]

**Comment by cimview** [comment]
Have you looked at the "60-Day SCED Disclosure Reports" (NP3-965-ER), or the 60-Dat DAM Disclosure" (NP3-966-ER) reports?  These are zip archives which contain generator resource data spreadsheets with HSL and Telemetered Resource Status.  These are released 60-days behind though, so you should be able to get January 2021 but the latest as of today is going to be around May 5, 2024.

**Comment by GlidePathAPI** [comment]
Not yet. Thank you for info!


Sean Ashman
Asset Management
C: 512.627.8966
Teams: 630.884.9154
***@***.******@***.***>
800 Town and Country Blvd, Suite 500
Houston, TX 77024

[A picture containing text, tableware, dishware  Description automatically generated]

**Comment by GlidePathAPI** [comment]
Thank you for pointing me in the right direction. I'm having trouble figuring out an efficient way to pull this data as I'm looking for about 3 years' worth of data. Currently working on a scrape, but this is all new to me so it's taking some time. Is it possible to request ERCOT provide this data via the API?


Sean Ashman
Asset Management
C: 512.627.8966
Teams: 630.884.9154
***@***.******@***.***>
800 Town and Country Blvd, Suite 500
Houston, TX 77024

[A picture containing text, tableware, dishware  Description automatically generated]

From: Sean Ashman
Sent: Friday, July 5, 2024 11:33
To: ercot/api-specs ***@***.***>; ercot/api-specs ***@***.***>
Cc: Author ***@***.***>
Subject: RE: [ercot/api-specs] ERCOT Public API Client - 60-Day COP All Updates (Discussion #40)

Not yet. Thank you for info!


Sean Ashman
Asset Management
C: 512.627.8966
Teams: 630.884.9154
***@***.******@***.***>
800 Town and Country Blvd, Suite 500
Houston, TX 77024

[A picture containing text, tableware, dishware  Description automatically generated]

**Comment by GlidePathAPI** [comment]
Afternoon -

We are currently in the process of implementing Pi Vision and looking to pipe in as much real time market data as possible. I'm having trouble finding the data listed in the URL below. Can you help me find if this data is available via API?

https://www.ercot.com/content/cdr/html/as_capacity_monitor.html



Sean Ashman
Asset Management
C: 512.627.8966
Teams: 630.884.9154
***@***.******@***.***>
800 Town and Country Blvd, Suite 500
Houston, TX 77024

[A picture containing text, tableware, dishware  Description automatically generated]

Glidepath Confidentiality Notice: The information transmitted by this email is intended only for the person or entity to which it is addressed. This email may contain proprietary, business-confidential and/or privileged material. If you are not the intended recipient of this message, be aware that any use, review, re-transmission, distribution, reproduction or any action taken in reliance upon this message is strictly prohibited. If you received this in error, please contact the sender and delete the material from all computers.

**Comment by GlidePathAPI** [comment]
Did anyone receive this?

Sean Ashman
Asset Management
C: 512.627.8966
Teams: 630.884.9154
***@***.******@***.***>
800 Town and Country Blvd, Suite 500
Houston, TX 77024

[A picture containing text, tableware, dishware  Description automatically generated]

From: Sean Ashman
Sent: Wednesday, November 6, 2024 11:29
To: ercot/api-specs ***@***.***>; ercot/api-specs ***@***.***>
Cc: Author ***@***.***>; Patrick Meyer ***@***.***>
Subject: ERCOT Public API Client - System Ancillary Service Capacity Monitor

Afternoon -

We are currently in the process of implementing Pi Vision and looking to pipe in as much real time market data as possible. I'm having trouble finding the data listed in the URL below. Can you help me find if this data is available via API?

https://www.ercot.com/content/cdr/html/as_capacity_monitor.html



Sean Ashman
Asset Management
C: 512.627.8966
Teams: 630.884.9154
***@***.******@***.***>
800 Town and Country Blvd, Suite 500
Houston, TX 77024

[A picture containing text, tableware, dishware  Description automatically generated]

Glidepath Confidentiality Notice: The information transmitted by this email is intended only for the person or entity to which it is addressed. This email may contain proprietary, business-confidential and/or privileged material. If you are not the intended recipient of this message, be aware that any use, review, re-transmission, distribution, reproduction or any action taken in reliance upon this message is strictly prohibited. If you received this in error, please contact the sender and delete the material from all computers.

**Reply by sscott-ercot** [reply]
The Ancillary Services Capacity Monitor report is not specifically available. We do offer 3 other AS reports, although probably not as frequent (ie. daily). 
 
2-Day Ancillary Services Reports
This report will contain all 48 Hour disclosure data related to DAM. The following individual files are included in the report:NP3-959-EX 48-hour Aggregate AS Offers; NP3-960-EX 48-hour Self-Arranged AS; NP3-961-EX 48-hour Cleared DAM AS (previously named 48 Hour Ancillary Services Reports).
 
DAM Ancillary Service Plan
Ancillary Service requirements by type and quantity for each hour of the current day plus the next 6 days.
 
Total Ancillary Service Offers
The total quantity in MW of Offers per Ancillary Service per hour from the Day-Ahead Market for the last thirty days on a daily basis which includes the following AS types: REGDN, REGUP, RRSPFR, RRSFFR, RRSUFR, & NSPIN.
